src/Tasks/Microsoft.NET.Build.Tasks/CheckForDuplicateItems.cs

```diff
@@ -27,11 +27,18 @@ public class CheckForDuplicateItems : TaskBase
         [Required]
         public string MoreInformationLink { get; set; }
 
+        [Output]
+        public ITaskItem [] DeduplicatedItems { get; set; }
+
         protected override void ExecuteCore()
         {
+            DeduplicatedItems = Array.Empty<ITaskItem>();
+
             if (DefaultItemsEnabled && DefaultItemsOfThisTypeEnabled)
             {
-                var duplicateItems = Items.GroupBy(i => i.ItemSpec).Where(g => g.Count() > 1).ToList();
+                var itemGroups = Items.GroupBy(i => i.ItemSpec);
```

Comment:
Actually I changed my mind. If we try to make this case insensitive, then even the implicit glob `**/*.cs` would fail if a user had both Class1.cs and class1.cs on Linux. 

I think the implication of leaving it case sensitive is that on Windows, a user could still get a duplicate sources error from compiler even with this fix. (E.g. if they have implicit `**/*.cs`, `<Compile Include="Class1.cs" />` and the file is actually named `class1.cs`)